indexer/services/roundtable/src/tasks/update-affiliate-info.ts

```diff
@@ -0,0 +1,63 @@
+import { logger } from '@dydxprotocol-indexer/base';
+import {
+  PersistentCacheTable,
+  AffiliateInfoTable,
+  PersistentCacheKeys,
+  PersistentCacheFromDatabase,
+  BlockFromDatabase,
+  BlockTable,
+  Transaction,
+  IsolationLevel,
+} from '@dydxprotocol-indexer/postgres';
+import { DateTime } from 'luxon';
+
+const defaultLastUpdateTime: string = '2024-09-16T00:00:00Z';
+
+/**
+ * Update the affiliate info for all affiliate addresses.
+ */
+export default async function runTask(): Promise<void> {
+  const latestBlock: BlockFromDatabase = await BlockTable.getLatest();
+  if (latestBlock.time === null) {
+    throw Error('Failed to get latest block time');
+  }
+
+  // Wrap getting cache, updating info, and setting cache in one transaction with row locking to
+  // prevent race condition on persistent cache rows between read and write.
+  const txId: number = await Transaction.start();
+  await Transaction.setIsolationLevel(txId, IsolationLevel.REPEATABLE_READ);
```

Comment:
This is the minimum isolation level to ensure that the row we read from persistent cache cannot be modified during the transaction. The side effect is that it also applies to the update query as well, which would lock all the rows in subaccounts belonging to affiliates. 

The entire roundtable should be fairly quick, but I still dont feel great about locking subaccounts. Thoughts?

An alternative solution is to explicity set the locking level in the query via
SELECT ... FOR UPDATE
but this would require more code change

Reply:
I thought about it more, I think we actually dont need the transaciton. Roundtable has logic to prevent mutliple of the same roundtable task from running at once and theres only one task editting this data right now. If we have two, then we'll have to worry about race conditions

Reply:
In that case, I will keep it in a transaction to enable rolling back in case any of the operations fail, but I will not set any isolation level.